KTB-15-creat-tg-buttons

DenisGubernatorov opened the pull request:
1) Регулярки для запросов вынесены за цикл
2) Класс Word перенесен в файл класса
3) В TelegramBotService создан метод sendMenu формирующий кнопки меню и отправляющий их боту 4) При получении команды "/start" от бота отправляется меню с кнопками 5) Перед циклом в Telegram.kt создается экземпляр LearnWordsTrainer

DenisGubernatorov committed:
KTB-15-creat-tg-buttons
1) Регулярки для запросов вынесены за цикл
2) Класс Word перенесен в файл класса
3) В TelegramBotService создан метод sendMenu формирующий кнопки меню и отправляющий их боту
4) При получении команды "/start" от бота отправляется меню с кнопками
5) Перед циклом в Telegram.kt создается экземпляр LearnWordsTrainer

dzenonce commented on src/main/kotlin/TelegramBotService.kt:
```diff
@@ -35,4 +35,41 @@ class TelegramBotService {
+            			[
+            				{
+            					"text": "Учить слова",
+            					"callback_data": "learn_word_button"
```
Давай вынесем `"learn_word_button"`  в константы `"satistics_button"` из тела `sendMenu`

dzenonce commented on src/main/kotlin/TelegramBotService.kt:
```diff
@@ -35,4 +35,41 @@ class TelegramBotService {
+            }
+            """.trimIndent()
+        val urlSendMessage = "$HOST_ADDRESS/bot$botToken/sendMessage"
+        val client: HttpClient = HttpClient.newBuilder().build()
```
`HttpClient` нужно сделать один общий для всего сервиса

dzenonce commented on src/main/kotlin/Telegram.kt:
```diff
@@ -1,33 +1,43 @@
 
 fun main(args: Array<String>) {
     val botToken: String = args[0]
     val botService = TelegramBotService()
```
Внутри `TelegramBotService` можно сделать поле `botToken`, чтобы не передавать в каждый метод а только один раз при создании.

dzenonce commented on src/main/kotlin/TelegramBotService.kt:
```diff
@@ -35,4 +35,41 @@ class TelegramBotService {
+        val sendMenuBody =
+            """
+            {
+            	"chat_id": ${chatId.toInt()},
```
Сейчас приложение крашится
При приведении к типу int, id пользователя может выходить из диапазона int и поэтому приложение падает с ошибкой
![errorLong](https://github.com/user-attachments/assets/23b7c403-b26c-4d78-a82c-1162bd4b981d)

Чтобы этого избежать, можно передавать просто строкой число или приводить к типу Long, но лучше передавать строкой

DenisGubernatorov committed:
KTB-15-creat-tg-buttons
1) Токен бота сделан полем в TelegramBotService и передается один раз в конструкторе, а не при вызове методов
2) Поле ИД чата передается в качестве строки, а не переводится в число
3) сallback_data для кнопок "Учить слова" и "Статистика" сделаны константами
4) HttpClient сделан свойством класса, а не создается отдельно в каждом методе

dzenonce reviewed (approved):
Все верно!
